mongo_connector/dest_mapping.py

```diff
@@ -0,0 +1,186 @@
+import logging
+import threading
+import re
+from mongo_connector import errors
+
+
+LOG = logging.getLogger(__name__)
+
+"""New structure to handle both plain and wildcard mapping namespaces dynamically.
+"""
+
+class DestMapping():
+    def __init__(self, namespace_set=[], ex_namespace_set=[], inpt={}):
+        # a dict containing plain mappings
+        self.plain = {}
+        # a dict containing wildcard mappings
+        self.wildcard = {}
+        # a dict containing reverse plain mapping
+        # because the mappings are not duplicated, so the values should also be unique
+        self.reverse_plain = {}
+        # a dict containing plain db mappings, db -> an array of mapped db
+        self.plain_db = {}
+        
+        # the input namespace_set and ex_namespace_set could contain wildcard
+        self.namespace_set = namespace_set
+        self.ex_namespace_set = ex_namespace_set
+        
+        self.lock = threading.Lock()
```

Comment:
Can you please explain why locking is necessary? It seems that the user should specify namespaces and the mapping before running the connector, then this object should remain immutable after that.